src/components/Modal/Modal.mdx

```diff
@@ -223,6 +226,63 @@ export const ModalStory = (args) => {
 
 <ArgTypes of={ModalStories.useModalStory} />
 
+## Confirmation Variant
+
+The "confirmation" `Modal` variant is used when the modal can confirm or revoke the modal's content. Accordingly, there are two buttons
+in the footer instead of the one in the default modal. Each button has its own custom function (`onCancel`, `onConfirm`) that will close the modal.
+The 'X' Close button in the upper right will call the `onCancel` function.
+
+Notes:
+
+- The `onConfirm`, `onCancel`, `confirmButtonLabel`, and `type` props are required for this variant.
+- The `onClose` prop should never be passed with `onConfirm` and `onCancel`.
+- When using the `ModalTrigger`, `onConfirm` and `onCancel` will call the `onClose` from the `useModal` hook internally.
+- When using the `useModal` hook, as with the `onClose` prop, pass the `onClose` function from the hook to `onCancel` and `onConfirm`:
+  for example, `onCancel: () => { onClose(); }`.
+
+<Source
+  code={`
+// Example for the Confirmation variant with ModalTrigger.
+  <ModalTrigger
+      buttonText="Confirm"
+      id="modal-confirmation"
+      modalProps={{
+        type: "confirmation",
+        bodyContent: "The action is happening",
+        cancelButtonLabel: "Cancel action",
+        confirmButtonLabel: "Confirm action",
+        onCancel: () => {
+          console.log("Custom cancel");
+        },
+        onConfirm: () => {
+          console.log("Custom confirm");
+        },
+        headingText: (
+          <Heading
+            level="h3"
+            display="flex"
+            alignItems="center"
+            gap="xs"
+            size="heading5"
+          >
+            <>
+              <Icon
+                color="ui.success.primary"
+                name="actionCheckCircleFilled"
+                size="large"
+              />
+              This is an action
+            </>
+          </Heading>
+        ),
+      }}
+    />
+`}
+  language="jsx"
+/>
+
+<Canvas of={ModalStories.ConfirmationVariant} />
```

Comment:
I see there are console messages being set for the button actions, but can the story be updated to include real-time updates onscreen?  If not, let's at least update the docs above to mention that users should take a look at the console as they are clicking the buttons.